### PR TITLE
feat: merge lowcodeconfig with registry, migrate alias config to demo

### PR DESCRIPTION
**Pull request by chilingling**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced theme support with dynamic theme imports based on configuration.
  - Added new configuration object for better customization and management.

- **Refactor**
  - Replaced `getGlobalConfig` with `getMergeMeta` across multiple components and hooks for improved configuration management.
  - Simplified alias assignments in Vite configuration.

- **Chores**
  - Removed outdated alias mappings and unnecessary imports.
  - Updated lifecycle configurations to be more modular and manageable.

### CODE REVIEW

File: packages/design-core/src/App.vue [hunk 1]
```diff
@@ -177,3 +182,4 @@ export default {
   }
 }
 </style>
+./theme/light.js
```
Comment (hexqi) [inline packages/design-core/src/App.vue]: 这行是多的？

File: packages/design-core/config/lowcode.config.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default {
     HTML: {},
     Flowchart: {}
   },
-
+  // TODO: 声明周期相关配置拆分到页面管理的配置项里面
```
Comment (hexqi) [inline packages/design-core/config/lowcode.config.js]: 这个文件可以删掉了

File: packages/design-core/src/init.js [hunk 3]
```diff
@@ -48,8 +46,8 @@ const defaultLifeCycles = {
       initMonitor(import.meta.env.VITE_ERROR_MONITOR_URL)
     }
 
-    window.TinyGlobalConfig = globalConfig
-    setGlobalConfig(globalConfig)
+    // 这里暴露到 window 是为了让 canvas 可以读取
+    window.TinyGlobalConfig = newRegistry.config || {}
```
Comment (hexqi) [inline packages/design-core/src/init.js:53]: 既然getGlobalConfig/setGlobalConfig都去掉了，那controller里面的globalconfig文件也删掉吧

File: packages/design-core/src/preview/src/App.vue [hunk 4]
```diff
@@ -4,10 +4,14 @@
...
 <script>
+import { getMergeMeta } from '@opentiny/tiny-engine-entry'
 import { useDebugSwitch } from './preview/debugSwitch'
 import Preview from './preview/Preview.vue'
 import Toolbar from './Toolbar.vue'
-import '@opentiny/tiny-engine-theme'
+
+const config = getMergeMeta('engine.config')
+
+import(`../../theme/${config.theme}.js`)
```
Comment (hexqi) [inline packages/design-core/src/preview/src/App.vue:14]: 这里有个问题就是，如果用户设计器单独开发了主题，配置使用外部的主题的话，这样就会引用不到报错了

Comment (chilingling) [reply packages/design-core/src/preview/src/App.vue:14]: 对的，用户自定义主题后续需要作为新特性进行支持。当前相关的PR有：https://github.com/opentiny/tiny-engine/pull/482
但是这个还没有进行评审

**Review by hexqi — APPROVED** [review]

File: packages/design-core/src/preview/src/App.vue (hunk 4 above)
Comment (hexqi) [reply packages/design-core/src/preview/src/App.vue:14]: 遗留

**Review by hexqi — APPROVED** [review]

**Pull request merged**